app/models/user.rb

```diff
@@ -257,6 +252,35 @@ class User < ApplicationRecord
 
   scope :eager_load_serialized_data, -> { includes(:roles) }
   scope :registered, -> { where(registered: true) }
+  # Unfortunately pg_search's default SQL query is not performant enough in this
+  # particular case (~ 500ms). There are multiple reasons:
+  # => creates a complex query like `SELECT FROM users INNER JOIN users` to compute ranking.
+  #    See https://github.com/Casecommons/pg_search/issues/292#issuecomment-202604151
+  # => it concatenates the content of `name` and the content of `username` to match
+  #    against the search term. By doing that, it can't use `tsvector` indexes correctly
+  #
+  # For these reasons we need to build a query manually using an `OR` condition,
+  # thus allowing the database to use the indexes properly. With this the SQL time is ~ 8-10ms.
+  #
+  # NOTE: we can't use unaccent() on the `tsvector` document because `unaccent()` can't be
+  # used in expression indexes as it's a mutable function and depends on server settings
+  # => https://stackoverflow.com/a/11007216/4186181
+  #
+  # An alternative solution: create a tsvector column and user a trigger as we do for articles.
```

Comment:
Given that going down from 500ms to 8-10ms seems more important, I think we should merge this as-is and regroup later if we want to add a trigger.

This is the exact behavior that there is right now in production:

- typing “João” will match with “João” but not with “Joao”
- typing “Joao” will match with “Joao” but not with “João”

to make those cases indistinguable, we need the trigger

Reply:
I'm good with the comment about _why_ we did it manually but I think we could remove the `An alternative solution:` part since that is not really relevant and that could just go in the PR description.